### PR TITLE
feat: Enable documentation mode Analyzer

### PR DESCRIPTION
**Pull request by ynsehoornenborg**
Make a separate Analyzer to warn the user that XML documentation Analyzers are not functional.

Created issue #439 for fixing the missing XML documentation.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -95,6 +95,9 @@ dotnet_diagnostic.CA2016.severity = error
 dotnet_diagnostic.CA2211.severity = error
 dotnet_diagnostic.CA2241.severity = error
 
+# Temporarly diabled to fix the tech dept.
```
Comment (bcollamore) [inline .editorconfig:98]: Temporar[i]ly di[s]abled

Comment (bcollamore) [reply .editorconfig:98]: Issue for this to re-enable?

File: Documentation/Diagnostics/PH2139.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# PH2139: Enable documentation creation
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2139 |
+| Category  | [Documentation](../Documentation.md) |
+| Analyzer | [EnableDocumentationCreationAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/EnableDocumentationCreationAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Error |
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2139.md:10]: I've been adding a row for "Enabled By Default"

File: Documentation/Diagnostics/PH2139.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Introduction
+
+When compileing code without creating XML documentation, the Analyzers don't get structed XML documentation and are unable to trigger diagnostics. This allows false negatives the occur.
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2139.md:14]: compiling

Comment (bcollamore) [inline Documentation/Diagnostics/PH2139.md:14]: the occur -> to occur

File: Documentation/Diagnostics/PH2139.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+## How to solve
+
+In .NET framework:
+Add the &lt;DocumentationFile&gt; element to the project file (.csproj), to be able to see Diagnostics for XML documentation.
```
Comment (bcollamore) [inline Documentation/Diagnostics/PH2139.md:19]: Consider adding a link to MS or other external documentation

File: Philips.CodeAnalysis.Common/DiagnosticId.cs [hunk 5]
```diff
@@ -131,5 +131,6 @@ public enum DiagnosticId
 		NamespaceMatchAssemblyName = 2135,
 		AvoidDuplicateStrings = 2136,
 		RegexNeedsTimeout = 3137,
+		EnableDocumentationCreation = 3139,
```
Comment (bcollamore) [inline Philips.CodeAnalysis.Common/DiagnosticId.cs:134]: 2139

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/EnableDocumentationCreationAnalyzer.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	}
+
+	public class EnableDocumentationCreationAction : SyntaxNodeAction<CompilationUnitSyntax>
+	{
+		private const int DiagnosticSpanStart = 0;
+		private const int DiagnosticSpanEnd = 3;
+
+		public override void Analyze()
+		{
+			if (Node.SyntaxTree.Options.DocumentationMode == DocumentationMode.None)
```
Comment (bcollamore) [inline Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/EnableDocumentationCreationAnalyzer.cs:30]: Is this going to squiggle the first 3 lines of every .cs file? I would propose no location.

**Review by bcollamore — CHANGES_REQUESTED** [review]

File: .editorconfig (hunk 1 above)
Comment (ynsehoornenborg) [reply .editorconfig:98]: Just too many violations, which would muddle this PR

File: Philips.CodeAnalysis.Common/DiagnosticId.cs (hunk 5 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.Common/DiagnosticId.cs:134]: Oops

File: Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/EnableDocumentationCreationAnalyzer.cs (hunk 6 above)
Comment (ynsehoornenborg) [reply Philips.CodeAnalysis.MaintainabilityAnalyzers/Documentation/EnableDocumentationCreationAnalyzer.cs:30]: I agree, location makes no sense here

File: Documentation/Diagnostics/PH2139.md (hunk 2 above)
Comment (ynsehoornenborg) [reply Documentation/Diagnostics/PH2139.md:10]: Added here also

File: .editorconfig (hunk 1 above)
Comment (ynsehoornenborg) [reply .editorconfig:98]: Issue #439 created

**Review by bcollamore — APPROVED** [review]

**Pull request merged**